local/node.go

```diff
@@ -38,27 +38,50 @@ type getConnFunc func(context.Context, node.Node) (net.Conn, error)
 // AvalancheGo binaries in tests
 type NodeProcess interface {
 	// Start this process
-	Start() error
+	Start(string, chan string) error
 	// Send a SIGTERM to this process
 	Stop() error
 	// Returns when the process finishes exiting
 	Wait() error
+	// Returns if the process is executing
+	Alive() bool
 }
 
 type nodeProcessImpl struct {
-	cmd *exec.Cmd
+	cmd              *exec.Cmd
+	running          bool
+	waitReturnCh     chan error
+	unexpectedStopCh chan string
 }
 
-func (p *nodeProcessImpl) Start() error {
-	return p.cmd.Start()
+func (p *nodeProcessImpl) Start(name string, unexpectedStopCh chan string) error {
+	p.unexpectedStopCh = unexpectedStopCh
+	startErr := p.cmd.Start()
+	p.running = true
+	p.waitReturnCh = make(chan error, 1)
+	go func() {
+		p.waitReturnCh <- p.cmd.Wait()
+		if p.running {
+			// Stop() was not called
+			p.running = false
+			p.unexpectedStopCh <- name
+		}
+	}()
+	return startErr
 }
 
 func (p *nodeProcessImpl) Wait() error {
-	return p.cmd.Wait()
+	return <-p.waitReturnCh
 }
 
 func (p *nodeProcessImpl) Stop() error {
```

Comment:
It might be nice for the user to be able to pass in a context, and if the context is cancelled but the process hasn't yet exited, we send a sigkill to ensure cleanup.

Reply:
I think this should be guarded by a `sync.Once` to make sure it only runs once. Idk what happens when you signal a dead process and it doesn't seem to be well documented.

Reply:
add context to Wait, and kill with SIGKILL process and descendants if context is cancelled

Reply:
ensured wait is executed once, and only called from certain node states. 
if you signal a dead linux process with syscall kill, you get int error for that https://man7.org/linux/man-pages/man2/kill.2.html 
I suppose that is propagated in https://pkg.go.dev/syscall#Kill but, ok, it is not documented how they mapped that to error type